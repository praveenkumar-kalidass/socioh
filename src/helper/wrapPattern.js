import React from 'react';

import { Background1 } from '../component/Background';

const wrapPattern = ({ Component }) => () => (
  <>
    <Component />
    <Background1 />
  </>
);

export default wrapPattern;
