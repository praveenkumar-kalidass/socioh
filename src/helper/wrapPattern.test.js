import React from 'react';
import { View } from 'react-native';

import wrapPattern from './wrapPattern';

describe('wrapPattern', () => {
  const mockComponent = () => <View />;

  it('should wrap with default pattern, when no pattern is specified', () => {
    const container = wrapPattern({
      Component: mockComponent,
    });

    expect(container).toMatchSnapshot();
  });

  it('should wrap with specific pattern, when pattern is specified', () => {
    const container = wrapPattern({
      Component: mockComponent,
      pattern: 'BACKGROUND_2',
    });

    expect(container).toMatchSnapshot();
  });
});
