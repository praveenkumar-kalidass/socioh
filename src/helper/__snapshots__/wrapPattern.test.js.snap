// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapPattern should wrap with default pattern, when no pattern is specified 1`] = `[Function]`;

exports[`wrapPattern should wrap with specific pattern, when pattern is specified 1`] = `[Function]`;
